ISSM: transition to GitHub repo

ISSM code was under subversion but is now hosted on GitHub

https://github.com/ISSMteam/ISSM

The spack package should be updated to use this as the code source

https://github.com/ACCESS-NRI/spack-packages/blob/main/packages/issm/package.py

Tried a test build of the current package with `spack/0.22`
```
spack install ism
```
The `scalapack` package failed to build with errors like
```
     160    /scratch/tm70/aph502/tmp/spack-stage/spack-stage-netlib-scalapack-2.2.0-hid6wnaqdo47dn3g4bzs6iuqlou56biv/spack-src/BLACS/
            SRC/igamx2d_.c: In function 'igamx2d_':                                                                                   
  >> 161    /scratch/tm70/aph502/tmp/spack-stage/spack-stage-netlib-scalapack-2.2.0-hid6wnaqdo47dn3g4bzs6iuqlou56biv/spack-src/BLACS/ 
            SRC/igamx2d_.c:201:7: error: implicit declaration of function 'BI_imvcopy'; did you mean 'BI_zmvcopy'? [-Wimplicit-functi
            on-declaration]                                                                                                                162      201 |       BI_imvcopy(Mpval(m), Mpval(n), A, tlda, bp->Buff);                                                         
     163          |       ^~~~~~~~~~                                                                                                 
     164          |       BI_zmvcopy                                                                                                 
  >> 165    /scratch/tm70/aph502/tmp/spack-stage/spack-stage-netlib-scalapack-2.2.0-hid6wnaqdo47dn3g4bzs6iuqlou56biv/spack-src/BLACS/
            SRC/igamx2d_.c:280:13: error: implicit declaration of function 'BI_ivmcopy'; did you mean 'BI_cvmcopy'? [-Wimplicit-funct
            ion-declaration]                                                                                                 
     166      280 |             BI_ivmcopy(Mpval(m), Mpval(n), A, tlda, bp2->Buff);                                                  
     167          |             ^~~~~~~~~~                                                                                           
     168          |             BI_cvmcopy                                                                           
```

According to this issue on the GitHub repo this happens with `gcc/14` and can be solved by building with `gcc/13`

https://github.com/Reference-ScaLAPACK/scalapack/issues/97

So I tested again with
```
spack install issm %gcc@13
```
which built `scalapack` without issue.

Unfortunately it stalled at the ISSM build because it needs a password for the svn server
```
[+] /g/data/tm70/aph502/spack/0.22/release/linux-rocky8-x86_64_v4/gcc-13.2.0/automake-1.16.5-3pzpcjcdrqpt3iaqy3zeqzqjybalbjtf
==> Installing issm-develop-olwa5wltn77pspqbxzptfedjet57v72s [28/28]
==> No binary for issm-develop-olwa5wltn77pspqbxzptfedjet57v72s found: installing from source
Authentication realm: <https://issm.ess.uci.edu:443> Your s383 passwd
Password for 'xxxxxx': 
```

Perhaps this was always an issue and required creds to be a available in a `.subversion` config file?

Will try modifying the package file and switching to GitHub source.

--- a/packages/issm/package.py
+++ b/packages/issm/package.py
@@ -27,9 +27,7 @@ class Issm(AutotoolsPackage):
     depends_on("m1qn3")
 
     def url_for_version(self, version):
-        url = "https://github.com/ISSMteam/ISSM/archive/refs/tags/v{0}.tar.gz".format(version)
-        return url
-
+        return "https://github.com/ISSMteam/ISSM/tarball/v{0}".format(version)
 
     def autoreconf(self, spec, prefix):
         autoreconf("--install", "--verbose", "--force")

--- a/packages/issm/package.py
+++ b/packages/issm/package.py
@@ -12,9 +12,10 @@ class Issm(AutotoolsPackage):
     """Ice-sheet and Sea-Level System Model"""
 
     homepage = "https://issm.jpl.nasa.gov/"
-    svn = "https://issm.ess.uci.edu/svn/issm/issm/trunk"
+    git = "https://github.com/ISSMteam/ISSM.git"
 
     version("develop")
+    version("4.24", sha256="0487bd025f37be4a39dfd48b047de6a6423e310dfe5281dbd9a52aa35b26151a")
 
     depends_on("autoconf", type="build")
     depends_on("automake", type="build")
@@ -24,6 +25,11 @@ class Issm(AutotoolsPackage):
     depends_on("mpi")
     depends_on("petsc+metis+mumps+scalapack")
     depends_on("m1qn3")
+
+    def url_for_version(self, version):
+        url = "https://github.com/ISSMteam/ISSM/archive/refs/tags/v{0}.tar.gz".format(version)
+        return url
+
 
     def autoreconf(self, spec, prefix):
         autoreconf("--install", "--verbose", "--force")

